src/mame/konami/konamigv.cpp

```diff
@@ -500,6 +544,17 @@ void tokimeki_state::machine_reset()
 	m_serial_sensor_id = 0;
 	m_serial_sensor_data = 0;
 	m_heartbeat_signal = 1;
+
+	m_printer_bit = 0;
+	m_printer_curbit = 0;
+	m_printer_curbyte = 0;
+	m_printer_is_printing = 0;
+	m_printer_current_image = 0;
+
+	std::fill(std::begin(m_printer_data), std::end(m_printer_data), 0);
+
+	m_page_bitmap.allocate(PRINTER_PAGE_WIDTH, PRINTER_PAGE_HEIGHT);
+	m_page_bitmap.fill(0xffffffff);
```

Comment:
Is it necessary to reallocate the bitmap here?  Could you just use `resize` rather than forcing the bitmap’s memory to be released and reallocated with `realloc`?

Reply:
I'm assuming you meant specifically the allocate line and not including the fill, right? The allocate can get moved to machine_start no problem.

Reply:
Yeah, the `allocate` line will cause it to delete the storage and reallocate it.  If you use `resize` instead, it won’t reallocate if it doesn’t need to.

(The `fill` isn’t a problem.  I selected both lines because you can’t select a single line for an in-place review comment, and if you don’t select two or more lines, you get some arbitrary number of lines quoted that will include stuff not relevant to the comment.)